greenmail-core/src/main/java/com/icegreen/greenmail/filestore/MBoxFileStore.java

```diff
@@ -0,0 +1,386 @@
+package com.icegreen.greenmail.filestore;
+
+import static com.icegreen.greenmail.imap.ImapConstants.HIERARCHY_DELIMITER;
+import static com.icegreen.greenmail.imap.ImapConstants.USER_NAMESPACE;
+import static java.nio.file.Files.readAllLines;
+import static java.nio.file.StandardOpenOption.CREATE;
+import static java.nio.file.StandardOpenOption.TRUNCATE_EXISTING;
+import static java.nio.file.StandardOpenOption.WRITE;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.StringTokenizer;
+import javax.mail.Quota;
+
+import com.icegreen.greenmail.configuration.GreenMailConfiguration;
+import com.icegreen.greenmail.imap.ImapConstants;
+import com.icegreen.greenmail.imap.ImapHostManager;
+import com.icegreen.greenmail.store.FolderException;
+import com.icegreen.greenmail.store.MailFolder;
+import com.icegreen.greenmail.store.Store;
+import com.icegreen.greenmail.user.GreenMailUser;
+import com.icegreen.greenmail.user.UserImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MBoxFileStore implements Store {
+
+    final Logger log = LoggerFactory.getLogger(MBoxFileStore.class);
+
+    private final Path rootDir;
+
+    private final Path userListFile;
+
+    private final Path pidFile;
+
+    private FileBaseContext ctx;
+
+
+    /**
+     * All classes implementing Store must have a public constructor which takes the configuration as parameter.
+     *
+     * @param startupConfig - startup configuration
+     */
+    public MBoxFileStore(GreenMailConfiguration startupConfig) {
+        this.rootDir = Paths.get(startupConfig.getFileStoreRootDirectory());
+        this.pidFile = this.rootDir.resolve("greenmail.pid");
+        if (Files.isRegularFile(this.pidFile)) {
```

Comment:
One of my goals when I program something is that when something goes wrong, then we should give the user as much information as possible to what was wrong, otherwise the user might loose a lot of time to desperately seek for what went wrong. And having two processes accessing the same store is IMHO such a thing which is very hard to find out when your tool 

a) does not support it
b) does not inform you about such a thing

And informing just by logging something as INFO is too weak, nobody reads INFO logging entries. But as we know that concurrent access to the same store does not work, we should really prevent this. This would be the fail-fast approach.

==> OK, I will drop the PID checking, even though I don't really agree with that this is not our business.